service-registry/src/main/java/org/apache/servicecomb/serviceregistry/client/IpPortManager.java

```diff
@@ -50,6 +50,10 @@
 
   private boolean autoDiscoveryInited = false;
 
+  public ArrayList<IpPort> getDefaultIpPort() {
+    return defaultIpPort;
```

Comment:
You can add a method such as getMaxRetryTimes based on available ips(not defaultIpPort), and minimum value is 2.